pkg/operation/botanist/resource_manager.go

```diff
@@ -0,0 +1,83 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/controlplane/resourcemanager"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DefaultResourceManager returns an instance of Gardener Resource Manager with defaults configured for being deployed in a Shoot namespace
+func (b *Botanist) DefaultResourceManager() (*resourcemanager.ResourceManager, error) {
+	image, err := b.ImageVector.FindImage(common.GardenerResourceManagerImageName, imagevector.RuntimeVersion(b.SeedVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
+	}
+
+	cfg := resourcemanager.Config{
+		AlwaysUpdate:    pointer.BoolPtr(true),
+		ConcurrentSyncs: pointer.Int32Ptr(20),
+		DefaultLabels: &map[string]string{
+			v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
+			v1beta1constants.LabelApp:   "gardener-resource-manager",
+		},
+		HealthSyncPeriod:           pointer.StringPtr("1m0s"),
+		MaxConcurrentHealthWorkers: pointer.Int32Ptr(10),
+		SyncPeriod:                 pointer.StringPtr("1m0s"),
+		TargetDisableCache:         pointer.BoolPtr(true),
+		WatchedNamespace:           pointer.StringPtr(b.Shoot.SeedNamespace),
+		// We run one GRM per shoot control plane, and the GRM is doing its leader election via configmaps in the seed -
+		// by default every 2s. This can lead to a lot of PUT /v1/configmaps requests on the API server, and given that
+		// a seed is very busy anyways, we should not unnecessarily stress the API server with this leader election.
+		// The GRM's sync period is 1m anyways, so it doesn't matter too much if the leadership determination may take up
+		// to one minute.
+		LeaseDuration: pointer.StringPtr("40s"),
+		RenewDeadline: pointer.StringPtr("15s"),
+		RetryPeriod:   pointer.StringPtr("10s"),
+	}
+
+	return resourcemanager.New(
+		b.K8sSeedClient.Client(),
+		b.Shoot.SeedNamespace,
+		image.String(),
+		b.Shoot.GetReplicas(1),
+		cfg,
+	), nil
+}
+
+// DeployGardenerResourceManager deploys the gardener-resource-manager
+func (b *Botanist) DeployGardenerResourceManager(ctx context.Context) error {
```

Comment:
Can you write a unit test for this function, similar to https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/kubescheduler_test.go#L80-L122?

Reply:
Done - though I think the test adds more of a burden than actual testing something meaningful.

Reply:
Maybe, yes. The function as of now is quite straight-forward. However, we have seen in the past that it might happen that such functions grow over time. If there was no initial test that can grow in parallel then it's likely that no tests are added at all or only very late (i.e., with additional effort/time consumption).